File: sally/programmers/week3/타겟 넘버.cpp
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;

int solution(vector<int> numbers, int target) {
    int answer = 0;
    int n = numbers.size();
    
    for (int i = 0; i < (1 << (n)); i++) {
        int tmp = 0;
        for (int j = 0; j < n; j++)
            if (i & (1 << j)) tmp += numbers[j]; // 1
            else tmp -= numbers[j]; // 0
        if (tmp == target) answer++;
    }

    return answer;
}

File: coodingpenguin/programmers/level2/타겟넘버.py
def solution(numbers, target):
    n = len(numbers)
    count = 0
    for i in range(1 << n):
        total = 0
        for j in range(n):
            # 비트가 1이면
            if i & 1 << j:
                total += numbers[j]  # 더해준다
            # 비트가 0이면
            else:
                total -= numbers[j]  # 빼준다
        # 결과가 타겟과 일치하면
        if total == target:
            count += 1  # 카운트를 +1 해준다
    return count
